### PR TITLE
feature: add default label to clusters

### PR DESCRIPTION
**Pull request by marinsalinas**

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_cluster.go [hunk 1]
```diff
@@ -675,7 +683,11 @@ func resourceMongoDBAtlasClusterUpdate(d *schema.ResourceData, meta interface{})
 		cluster.NumShards = pointy.Int64(cast.ToInt64(d.Get("num_shards")))
 	}
 	if d.HasChange("labels") {
-		cluster.Labels = expandLabelSliceFromSetSchema(d)
+		if containsLabelOrKey(expandLabelSliceFromSetSchema(d), defaultLabel) {
+			return fmt.Errorf("you should not set `Infraestructure Tool` label, it is used for internal purposes.")
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_cluster.go]: Infrastructure - spelling.

**Review by PacoDw — APPROVED** [review]
Nice! LGTM thank you

File: mongodbatlas/resource_mongodbatlas_cluster.go [hunk 2]
```diff
@@ -433,9 +435,14 @@ func resourceMongoDBAtlasClusterCreate(d *schema.ResourceData, meta interface{})
 		BiConnector:              biConnector,
 		ProviderSettings:         &providerSettings,
 		ReplicationSpecs:         replicationSpecs,
-		Labels:                   expandLabelSliceFromSetSchema(d),
 	}
 
+	if containsLabelOrKey(expandLabelSliceFromSetSchema(d), defaultLabel) {
+		return fmt.Errorf("you should not set `Infraestructure Tool` label, it is used for internal purposes.")
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_cluster.go]: Spelling, Infrastructure

File: website/docs/r/cluster.html.markdown [hunk 3]
```diff
@@ -233,6 +233,7 @@ Note free tier (M0) creation is not supported by the Atlas API and hence not sup
     The minimum disk size for dedicated clusters is 10GB for AWS and GCP. If you specify diskSizeGB with a lower disk size, Atlas defaults to the minimum disk size value.
 
 * `encryption_at_rest_provider` - (Optional) Set the Encryption at Rest parameter.  Possible values are AWS, GCP, AZURE or NONE.  Requires M10 or greater and for backup_enabled to be false or omitted.
+* `labels` - (Optional) Array containing key-value pairs that tag and categorize the cluster. Each key and value has a maximum length of 255 characters. You cannot set the key `Infraestructure Tool`, it is used for internal purposes.
```
Comment (themantissa) [inline website/docs/r/cluster.html.markdown]: Expanded just a little but glad you thought to include:
(Optional) Array containing key-value pairs that tag and categorize the cluster. Each key and value has a maximum length of 255 characters. You cannot set the key `Infrastructure Tool`, it is used for internal purposes to track aggregate usage.

File: mongodbatlas/resource_mongodbatlas_cluster.go [hunk 4]
```diff
@@ -33,6 +33,8 @@ const (
 	errorAdvancedConfRead   = "error reading Advanced Configuration Option form MongoDB Cluster (%s): %s"
 )
 
+var defaultLabel = matlas.Label{Key: "Infraestructure Tool", Value: "MongoDB Atlas Terraform Provider"}
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_cluster.go]: Infrastructure spelling

File: mongodbatlas/provider.go [hunk 5]
```diff
@@ -171,6 +172,31 @@ func expandLabelSliceFromSetSchema(d *schema.ResourceData) []matlas.Label {
...
+			}
+			continue
+		}
+		pos++
+
+	}
+	return list
+}
+
 func getPluginVersion() string {
```
Comment (themantissa) [inline mongodbatlas/provider.go]: I think we can remove this func now?

File: mongodbatlas/provider_test.go [hunk 6]
```diff
@@ -121,3 +122,26 @@ func TestDecodeID(t *testing.T) {
 		t.Fatalf("Bad TestDecodeID return \n got = %#v\nwant = %#v \ndiff = %#v", got, got2, diff)
 	}
 }
+
```
Comment (themantissa) [inline mongodbatlas/provider_test.go:117]: Did we do a manual test to check that Infrastructure Tool is getting set?

**Review by themantissa — CHANGES_REQUESTED** [review]
A few small changes/questions but looking good

File: mongodbatlas/provider.go (hunk 5 above)
Comment (marinsalinas) [reply mongodbatlas/provider.go]: Which function? getPluginVersion()?

File: mongodbatlas/provider_test.go (hunk 6 above)
Comment (marinsalinas) [reply mongodbatlas/provider_test.go:117]: I did it, you can try as well, creating a cluster without labels, see the GUI.

File: mongodbatlas/resource_mongodbatlas_cluster.go (hunk 4 above)
Comment (marinsalinas) [reply mongodbatlas/resource_mongodbatlas_cluster.go]: Got let gonna change.

File: mongodbatlas/resource_mongodbatlas_cluster.go (hunk 2 above)
Comment (marinsalinas) [reply mongodbatlas/resource_mongodbatlas_cluster.go]: Got it gonna change

File: mongodbatlas/provider.go (hunk 5 above)
Comment (themantissa) [reply mongodbatlas/provider.go]: Yup, getPluginVersion.

File: mongodbatlas/provider_test.go (hunk 6 above)
Comment (themantissa) [reply mongodbatlas/provider_test.go:117]: Just wanted to make sure - not sure what you mean see GUI?

File: mongodbatlas/provider.go (hunk 5 above)
Comment (marinsalinas) [reply mongodbatlas/provider.go]: is it no longer needed?

File: mongodbatlas/provider_test.go (hunk 6 above)
Comment (marinsalinas) [reply mongodbatlas/provider_test.go:117]: Sorry, I wanted to say the DEBUG Logs, 

run the apply with debug option - `TF_LOG=DEBUG  terraform apply --auto-approve`

There you can see the label when create and read the cluster.

File: mongodbatlas/provider.go (hunk 5 above)
Comment (themantissa) [reply mongodbatlas/provider.go]: That's what I'm asking you.  If we aren't able to write the version to the label then it makes sense to remove it.

File: mongodbatlas/provider_test.go (hunk 6 above)
Comment (themantissa) [reply mongodbatlas/provider_test.go:117]: Makes sense. Thank you!

File: mongodbatlas/provider.go (hunk 5 above)
Comment (marinsalinas) [reply mongodbatlas/provider.go]: got it, let me remove it.

Comment (marinsalinas) [reply mongodbatlas/provider.go]: got it, I'll remove it.

**Review by themantissa — APPROVED** [review]
LGTM!

**Pull request merged**